File: day_01/day01.rb
input = File.read(File.join(File.dirname(__FILE__),'input.txt')).strip

class Day01

  def self.captcha_next captcha
    nums = captcha.split('')
    nums.select.with_index do |x, i|
      nums[(i + 1) % nums.length] == x
    end.map(&:to_i).reduce(0, :+)
  end

  def self.captcha_half captcha
    nums = captcha.split('')
    nl = nums.length
    nums.select.with_index do |x, i|
      nums[(i + nl / 2) % nl] == x
    end.map(&:to_i).reduce(0, :+)
  end

end

p "Part 1 answer is #{Day01.captcha_next input}"
p "Part 2 answer is #{Day01.captcha_half input}"

File: day_01/day01.py
import os

with open(os.path.join(os.path.dirname(__file__), 'input.txt'), 'r') as inputfile:
    data = inputfile.read().replace("\n","")

def captcha_next(captcha):
    nums  = list(map(int, captcha))
    nl = len(nums)
    total = 0
    for index, num in enumerate(nums):
        if nums[(index + 1) % nl] == num:
            total += num
    return total

def captcha_half(captcha):
    nums  = list(map(int, captcha))
    nl = len(nums)
    total = 0
    for index, num in enumerate(nums):
        if nums[int((index + nl / 2) % nl)] == num:
            total += num
    return total

print(f'Part 1 answer is {captcha_next(data)}')
print(f'Part 2 answer is {captcha_half(data)}')
